code/FsComposite/Composite.Core/Processing.fs

```diff
@@ -21,5 +21,24 @@ module Processing =
         | x -> f_transform x
 
     let find_2_and_transform_strict f_is_1 f_is_2 f_transform lst =
-        let f_transform_strict = function | (Some x1, Some x2) -> l(f_transform (x1, x2)) | _ ->  LazyList.empty
+        let f_transform_strict = function | (Some x1, Some x2) -> ll(f_transform (x1, x2)) | _ ->  LazyList.empty

```

Comment:
The old functions are out of place here. The new ones should be renamed to something like 'cata' etc. The old ones should be removed. We are moving towards generic folds.